src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+const LiElement = tree.querySelectorAll('li');
+
+LiElement.forEach((listItem) => {
+  const ulElement = listItem.querySelector('ul');
+
+  if (ulElement) {
+    const span = document.createElement('span');
+    const text = listItem.firstChild;
```

Comment:
The variable `text` is assigned `listItem.firstChild`, which may not always be a text node. Consider checking if `text` is indeed a text node before accessing `textContent`.